### PR TITLE
#171789422 user can view their requests

### PR DESCRIPTION
**Pull request by Baraka-Mugisha**
#### What does this PR do?
- add a feature whereby a user can view his requests

#### Description of Task to be completed?
- added the feature's controller
- tested the feature
- documents the feature with swagger
- internationalized response messages
- modified notification links

#### How should this be manually tested?
- clone this [repo](https://github.com/andela/devrepublic-bn-backend.git)
- run npm install to install all dependencies
- run the server (npm run reset:dev)
- make requests
- make a get method to `/api/v1/trips/{requestId}/view` to view one specific request you made
- or make a get method to `/api/v1/trips/view` to view all requests you made

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- [#171789422](https://www.pivotaltracker.com/story/show/171789422)
- [#171789813](https://www.pivotaltracker.com/story/show/171789813)

#### Screenshots (if appropriate)
Screen Shot 2020-03-19 at 13 58 42

Screen Shot 2020-03-19 at 13 59 21

#### Questions:
- N/A

### CODE REVIEW

File: src/controllers/tripsController.js [hunk 1]
```diff
@@ -373,4 +359,67 @@ export default class requestController {
...
+        const availableRequest = await db.Request.findOne({
+          where: {
+            id: requestId,
+            managerId: user.id
+          }
+        });
+        if (!availableRequest) {
+          return Response.errorResponse(res, 404, res.__('Request not found or not yours to manage'));
+        }
+        return Response.success(res, 200, res.__('Request found'), availableRequest);
```
Comment (izzett222) [inline src/controllers/tripsController.js]: hey, @Baraka-Mugisha did you test for when a manager wants to view an individual request?

**Review by izzett222 — CHANGES_REQUESTED** [review]

**Review by Ntare22 — CHANGES_REQUESTED** [review]
Could you change the PR heading to match others?

File: src/controllers/tripsController.js (hunk 1 above)
Comment (Baraka-Mugisha) [reply src/controllers/tripsController.js]: I added the test

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

File: src/controllers/tripsController.js [hunk 2]
```diff
@@ -428,4 +431,65 @@ export default class requestController {
...
+   */
+  static async viewAllRequests(req, res) {
+    try {
+      const { user } = req;
+      if (user.role === 'manager') {
+        const availableRequests = await db.Request.findAll({ where: { managerId: user.id } });
+        return availableRequests.length === 0
+          ? Response.errorResponse(res, 404, res.__('No trip requests available'))
+          : Response.success(res, 200, res.__('Pending requests to approve'), availableRequests);
+      }
```
Comment (izzett222) [inline src/controllers/tripsController.js]: the find query above returns all requests whether open or rejected or rejected but the response message is saying that it returning only open requests.

**Review by izzett222 — APPROVED** [review]

File: src/swagger/trips.swagger.js [hunk 3]
```diff
@@ -436,3 +440,59 @@
...
+/**
+ * @swagger
+ * /api/v1/trips/{requestId}/view:
+ *   get:
+ *     security:
+ *       - bearerAuth: []
+ *     tags:
+ *       - Trips
+ *     name: User view a trip request
+ *     summary:  User should be able to view a trip request he made
```
Comment (izzett222) [inline src/swagger/trips.swagger.js]: doesn't the manager also view a request made by someone in their direct report

**Review by izzett222 — CHANGES_REQUESTED** [review]

File: src/swagger/trips.swagger.js (hunk 3 above)
Comment (Baraka-Mugisha) [reply src/swagger/trips.swagger.js]: I added the description

File: src/controllers/tripsController.js [hunk 4]
```diff
@@ -1,11 +1,14 @@
...
+import dotenv from 'dotenv';
 import db from '../models';
 import Response from '../utils/ResponseHandler';
 import TripsService from '../services/tripServices';
 import stringHelper from '../utils/stringHelper';
 import notifService from '../services/notificationService';
 import SendNotification from '../utils/sendNotification';
 
+dotenv.config();
+
```
Comment (bbaime98) [inline src/controllers/tripsController.js]: do we need this dotenv?

**Review by bbaime98 — CHANGES_REQUESTED** [review]

File: src/controllers/tripsController.js (hunk 4 above)
Comment (Baraka-Mugisha) [reply src/controllers/tripsController.js]: I needed it to update the notification link, but now, it is no longer needed

**Review by bbaime98 — DISMISSED** [review]

**Review by Ntare22 — DISMISSED** [review]

**Review by izzett222 — DISMISSED** [review]

**Review by Ntare22 — APPROVED** [review]

**Review by bbaime98 — APPROVED** [review]

**Review by izzett222 — APPROVED** [review]

**Pull request merged**